### ISSUE TITLE
setHeader uses proprietary templating for SpEL

### ISSUE THREAD

**Issue by t-burch** [membrane/api-gateway#1055]
The setHeader interceptor uses `${<SpEL>}` as its template for SpEL expressions, but why not use the actual `#{<SpEL>}` template so it integrates with the SpEL context given in the proxies.xml file?

**Comment by predic8** [comment]
Good point. We have lots of samples, we should therefore move slowy. @t-burch can you support both #{} and ${}. Can you also rewrite the setHeader examples?

In 6.0.0 we can remove the support for ${}.

**Comment by t-burch** [comment]
No examples use setHeader at the moment, so the only task would be to remove $ once 6.0 rolls out
